nanovdb/nanovdb/NanoVDB.h

```diff
@@ -529,92 +742,93 @@ __hostdev__ inline static const T* alignPtr(const T* p)
-        : mData{{(uint8_t(0.5f + r * 255.0f)), // round to nearest
-                 (uint8_t(0.5f + g * 255.0f)), // round to nearest
-                 (uint8_t(0.5f + b * 255.0f)), // round to nearest
-                 (uint8_t(0.5f + a * 255.0f))}}// round to nearest
-    {
-    }
-    __hostdev__ bool operator<(const Rgba8& rhs) const { return mData.packed < rhs.mData.packed; }
-    __hostdev__ bool operator==(const Rgba8& rhs) const { return mData.packed == rhs.mData.packed; }
-    __hostdev__ float lengthSqr() const
-    {
-        return 0.0000153787005f*(float(mData.c[0])*mData.c[0] +
-                                 float(mData.c[1])*mData.c[1] +
-                                 float(mData.c[2])*mData.c[2]);//1/255^2
-    }
-    __hostdev__ float length() const { return sqrtf(this->lengthSqr() ); }
-    __hostdev__ const uint8_t& operator[](int n) const { return mData.c[n]; }
-    __hostdev__ uint8_t& operator[](int n) { return mData.c[n]; }
-    __hostdev__ const uint32_t& packed() const { return mData.packed; }
-    __hostdev__ uint32_t& packed() { return mData.packed; }
-    __hostdev__ const uint8_t& r() const { return mData.c[0]; }
-    __hostdev__ const uint8_t& g() const { return mData.c[1]; }
-    __hostdev__ const uint8_t& b() const { return mData.c[2]; }
-    __hostdev__ const uint8_t& a() const { return mData.c[3]; }
-    __hostdev__ uint8_t& r() { return mData.c[0]; }
-    __hostdev__ uint8_t& g() { return mData.c[1]; }
-    __hostdev__ uint8_t& b() { return mData.c[2]; }
-    __hostdev__ uint8_t& a() { return mData.c[3]; }
-};// Rgba8
-
-using PackedRGBA8 = Rgba8;// for backwards compatibility
+// --------------------------> isFloatingPointVector(GridType) <------------------------------------
 
-// --------------------------> isValue(GridType, GridClass) <------------------------------------
+/// @brief return true if the GridType maps to a floating point vec3.
+__hostdev__ inline bool isFloatingPointVector(GridType gridType)
+{
+    return gridType == GridType::Vec3f ||
+           gridType == GridType::Vec3d ||
+           gridType == GridType::Vec4f ||
+           gridType == GridType::Vec4d;
+}
 
-/// @brief return true if the GridType maps to a floating point value.
-__hostdev__ inline bool isFloatingPoint(GridType gridType)
+// --------------------------> isInteger(GridType) <------------------------------------
+
+/// @brief return true if the GridType maps to a index type.
+__hostdev__ inline bool isInteger(GridType gridType)
 {
-    return gridType == GridType::Float  ||
-           gridType == GridType::Double ||
-           gridType == GridType::Fp4    ||
-           gridType == GridType::Fp8    ||
-           gridType == GridType::Fp16   ||
-           gridType == GridType::FpN;
+    return gridType == GridType::Int16 ||
+           gridType == GridType::Int32 ||
+           gridType == GridType::Int64 ||
+           gridType == GridType::UInt32;
```

Comment:
I'm not sure if `GridType::Index`, `GridType::OnIndex`, `GridType::IndexMask`, `GridType::OnIndexMask` should also be included here. I think they belong to another 'semantic'.

Reply:
The types you mention are handled by isIndex defined below. Since these index grids don't actually store anything per voxel (not even an integer) they are treated differently than for instance GridType::Int32 that actually encodes a single int per voxel. In other words, these methods reflect both the implementation as well as the return type. So, long story short it's correct as is but I  do agree that the documentation is confusing so I'll fix that instead :)

Reply:
pushed up improved documentation